src/devices/machine/spi_sdcard.h

```diff
@@ -35,25 +35,41 @@ class spi_sdcard_device : public device_t
 
 	required_device<harddisk_image_device> m_image;
 
-	int m_type;
+	sd_type m_type;
 
 private:
-	enum
+	enum sd_state : u8
 	{
+		//REF Table 4-1:Overview of Card States vs. Operation Mode
 		SD_STATE_IDLE = 0,
+		SD_STATE_READY,
+		SD_STATE_IDENT,
+		SD_STATE_STBY,
+		SD_STATE_TRAN,
+		SD_STATE_DATA,
+		SD_STATE_DATA_MULTI, // synthetical state for this implementation
+		SD_STATE_RCV,
+		SD_STATE_PRG,
+		SD_STATE_DIS,
+		SD_STATE_INA,
+
+		//FIXME Existing states wich must be revisited
 		SD_STATE_WRITE_WAITFE,
 		SD_STATE_WRITE_DATA
-	};
+	} m_state;
 
-	void send_data(int count);
-	void do_command();
+	void send_data(u16 count, sd_state new_state);
+	void do_command(u8 m_cmd[6]);
```

Comment:
This has a parameter name that shadows a class data member name – that’s generally undesirable.

Reply:
The issue is still here – this overload of `do_command` has a formal parameter called `m_cmd` that shadows the data member `m_cmd`.   It’s also a bad idea to use the `m_` prefix for anything other than instance data members, as conventionally it’s used as a prefix to mark them.